va/va_test.go

```diff
@@ -1052,12 +1014,27 @@ func TestLimitedReader(t *testing.T) {
 		"Expected failure due to truncation")
 }
 
-func setup() (*ValidationAuthorityImpl, *mocks.Statter, *blog.Mock) {
+func setup(t *testing.T, srv *httptest.Server) (*ValidationAuthorityImpl, int, *mocks.Statter, *blog.Mock) {
 	stats := mocks.NewStatter()
 	scope := metrics.NewStatsdScope(stats, "VA")
 	logger := blog.NewMock()
+
+	var portConfig cmd.PortConfig
+	if srv != nil {
+		port, err := getPort(srv)
+		if err != nil {
+			// We never expect to fail to get the port for an http server, so treat it
+			// as a fatal test failure and halt immediately
```

Comment:
I was actually suggesting to move this assertion into `getPort` itself, so `getPort` panics if it sees any errors. That would simplify all the call sites for `getPort`. That also lets us get rid of the `t` param to this function.